### PR TITLE
#13: [Form editor] Drag to reorder form fields	

### PR DESCRIPTION
**Pull request by DaoyuT**
Need help:
- [x] ~~drag reorder works and saves the data, but row shows blank when dragging, how to debug?~~ css below
- [x] `npm install -s` https -> http, keep?

TODO:
- [x] make handle prettier 
- [x] css: change row to be on top of table

### CODE REVIEW

File: web/package-lock.json [hunk 1]
```diff
@@ -1035,7 +1035,7 @@
         },
         "whatwg-fetch": {
           "version": "2.0.4",
-          "resolved": "https://registry.npmjs.org/whatwg-fetch/-/whatwg-fetch-2.0.4.tgz",
+          "resolved": "http://registry.npmjs.org/whatwg-fetch/-/whatwg-fetch-2.0.4.tgz",
```
Comment (gino-m) [inline web/package-lock.json]: Does https://github.com/npm/npm/issues/20106#issuecomment-383770315 help?

Looks like we may need to upgrade to a newer version of nvm to permanently fix the problem. Please lmk if it works?

File: web/src/components/gnd-feature-type-editor/gnd-form-editor.js [hunk 2]
```diff
@@ -48,6 +53,23 @@ const styles = (theme) => ({
   },
 });
 
+const DragHandle = sortableHandle(() => <span>::</span>);
```
Comment (gino-m) [inline web/src/components/gnd-feature-type-editor/gnd-form-editor.js]: For this I think we can use `drag_indicator` from https://www.npmjs.com/package/@material-ui/icons

File: web/src/components/gnd-feature-type-editor/gnd-form-editor.js [hunk 3]
```diff
@@ -48,6 +53,23 @@ const styles = (theme) => ({
...
+    <DragHandle />
+    <GndFormElementEditor
+      key={element.id}
+      element={element}
+      onChange={onChange}
+    />
+  </div>
+));
+
+const SortableContainer = sortableContainer(({children}) => {
```
Comment (gino-m) [inline web/src/components/gnd-feature-type-editor/gnd-form-editor.js]: According to [the docs](https://github.com/clauderic/react-sortable-hoc#sortablecontainer-hoc), we can also set `lockAxis` to `y` to ensure dragging happens only vertically.

File: web/src/components/gnd-feature-type-editor/gnd-form-editor.js [hunk 4]
```diff
@@ -48,6 +53,23 @@ const styles = (theme) => ({
   },
 });
 
+const DragHandle = sortableHandle(() => <span>::</span>);
+
+const SortableItem = sortableElement(({element, onChange}) => (
```
Comment (gino-m) [inline web/src/components/gnd-feature-type-editor/gnd-form-editor.js]: You can try setting the z-index on a helper as in https://github.com/clauderic/react-sortable-hoc/issues/87#issuecomment-405911683 to make sure the item stays above the modal while dragging.

File: web/package-lock.json (hunk 1 above)
Comment (DaoyuT) [reply web/package-lock.json]: It worked, thanks!

**Pull request merged**